File: Flatter Project/sorce code/sms_by_flutter/lib/model/syllabus_model.dart
class SyllabusModel {
  int? syid;
  String? sclass;
  String? examCategory;
  String? subject;
  String? pageNo;
  String? description;
  SubjectModel? subjectModel;

  SyllabusModel({
    this.syid,
    this.sclass,
    this.examCategory,
    this.subject,
    this.pageNo,
    this.description,
    this.subjectModel,
  });

  factory SyllabusModel.fromJson(Map<String, dynamic> json) {
    return SyllabusModel(
      syid: json['syid'],
      sclass: json['sclass'],
      examCategory: json['examCatagory'],
      subject: json['subject'],
      pageNo: json['pageNo'],
      description: json['discription'],
      subjectModel: SubjectModel.fromJson(json['subjectModel']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'syid': syid,
      'sclass': sclass,
      'examCatagory': examCategory,
      'subject': subject,
      'pageNo': pageNo,
      'discription': description,
      'subjectModel': subjectModel?.toJson(),
    };
  }
}

class SubjectModel {
   int? subid;
   String? subName;

  SubjectModel({
    this.subid,
    this.subName,
  });

  factory SubjectModel.fromJson(Map<String, dynamic> json) {
    return SubjectModel(
      subid: json['subid'],
      subName: json['subName'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'subid': subid,
      'subName': subName,
    };
  }
}

File: Flatter Project/sorce code/SMS-With-Maven/src/main/java/com/mdfaysalhossain/SMS/With/Maven/model/SyllabusModel.java
package com.mdfaysalhossain.SMS.With.Maven.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Table(name = "syllabusTable")

public class SyllabusModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long syid;

    private String sclass;

    private String examCatagory;


    @Column(unique = true)
    private String subject;

    private String pageNo;

    private String discription;


    @ManyToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "subject_id")
    private SubjectModel subjectModel;
}
